### PR TITLE
feat(dropdown): auto and array support in placement

### PR DESCRIPTION
**Pull request by mschoudry**
Added support for auto placement and array can be defined inside placement attribute of dropdown.

### CODE REVIEW

File: src/dropdown/dropdown.spec.ts [hunk 1]
```diff
@@ -515,6 +515,42 @@ describe('ngb-dropdown-toggle', () => {
...
+          </div>
+      </div>`;
+
+    const fixture = createTestComponent(html);
+    const compiled = fixture.nativeElement;
+
+    const dropDownEle = getDropdownEl(compiled);
+
+    expect(dropDownEle).toHaveCssClass('dropdown');
+    expect(dropDownEle.getAttribute('ng-reflect-placement')).toEqual('auto');
```
Comment (pkozlowski-opensource) [inline src/dropdown/dropdown.spec.ts]: We should not use `ng-reflect-` attributes in tests as it is impl-specific and can change at any moment (not to mention the fact that it is not there in production mode).

TBH I'm not sure what you are trying to assert with this test so it is hard for me to suggest changes...

File: src/dropdown/dropdown.spec.ts [hunk 2]
```diff
@@ -515,6 +515,42 @@ describe('ngb-dropdown-toggle', () => {
...
+          </div>
+      </div>`;
+
+    const fixture = createTestComponent(html);
+    const compiled = fixture.nativeElement;
+
+    const dropDownEle = getDropdownEl(compiled);
+
+    expect(dropDownEle).toHaveCssClass('dropdown');
+    expect(dropDownEle.getAttribute('ng-reflect-placement')).toEqual('left-top,top-left,auto');
```
Comment (pkozlowski-opensource) [inline src/dropdown/dropdown.spec.ts]: Same comment as above...

File: src/dropdown/dropdown.ts [hunk 3]
```diff
@@ -10,20 +10,23 @@ import {
...
 import {NgbDropdownConfig} from './dropdown-config';
-import {positionElements} from '../util/positioning';
+import {positionElements, PlacementArray, Placement} from '../util/positioning';
 
 /**
  */
 @Directive(
     {selector: '[ngbDropdownMenu]', host: {'[class.dropdown-menu]': 'true', '[class.show]': 'dropdown.isOpen()'}})
 export class NgbDropdownMenu {
+  placement: Placement = 'bottom';
```
Comment (pkozlowski-opensource) [inline src/dropdown/dropdown.ts]: Why not `PlacementArray`?

Comment (pkozlowski-opensource) [reply src/dropdown/dropdown.ts]: Oh, I see, sorry for the noise.

File: src/dropdown/dropdown.ts [hunk 4]
```diff
@@ -102,9 +108,17 @@ export class NgbDropdown {
     this._zoneSubscription = ngZone.onStable.subscribe(() => { this._positionMenu(); });
   }
 
-  isUp() { return this.placement.indexOf('top') !== -1; }
+  isUp() {
```
Comment (pkozlowski-opensource) [inline src/dropdown/dropdown.ts]: Maybe we should simplify those methods in the following way:
* `isDown()` is true only when there is `bottom` somewhere in placement
* `isUp()` = `!isDown()`

WDYT?

In any case we should tests for this logic.

**Comment by jbistis** [comment]
Is there a fix for this?

**Comment by suhailkc** [comment]
Is there any fix for this ???

**Comment by mikelhamer** [comment]
A fix would be awesome :)

**Pull request closed without merging**